.github/workflows/ci_tests.yaml

```
@@ -0,0 +1,100 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  # Schedule daily tests
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  test:
+    name: ${{ matrix.os }} - Python ${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.7]
+        os: [ubuntu-latest, macOS-latest, windows-latest]
+
+    steps:
+      # Checkout current git repository
+      - name: Checkout
+        uses: actions/checkout@v2.1.0
+
+      # Workaround for versioneer, see https://github.com/warner/python-versioneer/issues/206
+      - name: Fetch all history for all tags and branches
+        run: git fetch --prune --unshallow
+
+      # Setup Miniconda
+      - name: Setup Miniconda
+        uses: goanpeca/setup-miniconda@v1.3.1
+        with:
+          python-version: ${{ matrix.python-version }}
+          channels: conda-forge
+
+      # Install GMT and other required dependencies from conda-forge
+      - name: Install GMT and required dependencies
+        shell: bash -l {0}
+        run: |
+          requirements_file=full-conda-requirements.txt
+          cat requirements.txt requirements-dev.txt > $requirements_file
+          cat << EOF >> $requirements_file
+          gmt=6.0.0
+          make
+          EOF
+          conda install --yes --file $requirements_file
+
+      # Show installed pkg information for postmortem diagnostic
+      - name: List installed packages
+        shell: bash -l {0}
+        run: conda list
+
+      # Cache the ${HOME}/.gmt directory, for docs and testing
+      - name: Cache GMT directory
+        uses: actions/cache@v2
+        id: cache
+        with:
+          path: |
+            ~/.gmt/cache
+            ~/.gmt/server
+          key: cache-gmt-${{ runner.os }}-20200609
```

```suggestion
          key: cache-gmt-${{ github.ref }}-${{ runner.os }}-20200609
```
Sorry, should actually include the github ref here (either master, or the pull request number).

---

I don't get it. Does ` key: cache-gmt-${{ github.ref }}-${{ runner.os }}-20200609` mean that each branch and PRs will have different caches?

---

Short answer: Yes. Long answer: That's to ensure that the PR cache doesn't 'pollute' the master cache. If you look at the next line `restore-keys: cache-gmt-refs/heads/master-`, it means that each PR can restore from the master's cache, but the master branch cannot see anything from the PRs. See also https://github.com/manubot/rootstock/pull/316/files#r386037243 where I got the idea from.

---

The PR cache should never be allowed to pollute the master cache. If it can, then it's a bad design of the cache mechanism. 

Anyway, it looks no harm to have `github.ref` in it.